projects/plugins/social/src/js/components/admin-page/index.jsx

```diff
@@ -63,13 +66,49 @@ const ConnectionItems = () => {
 		);
 	} );
 };
+
+const ModuleToggle = () => {
+	const updateOptions = useDispatch( STORE_ID ).updateJetpackSettings;
+	const isModuleEnabled = useSelect( select => select( STORE_ID ).isModuleEnabled() );
+	const isUpdating = useSelect( select => select( STORE_ID ).isUpdatingJetpackSettings() );
+
+	const toggleModule = useCallback( () => {
+		const newOption = {
+			publicize_active: ! isModuleEnabled,
+		};
+		updateOptions( newOption );
+	}, [ isModuleEnabled, updateOptions ] );
+
+	const label = isModuleEnabled
+		? __( 'Jetpack Social is active', 'jetpack-social' )
+		: __( 'Jetpack Social is inactive', 'jetpack-social' );
+
+	return (
+		<ToggleControl
+			label={ __( 'Activate Jetpack Social', 'jetpack-social' ) }
+			help={ isUpdating ? __( 'Updating…', 'jetpack-social' ) : label }
+			disabled={ isUpdating }
+			checked={ isModuleEnabled }
+			onChange={ toggleModule }
+		/>
+	);
+};
+
 const Admin = () => {
+	useMemo( () => {
```

Comment:
`useMemo` is usually used for expensive computations while as `useEffect` can be used for side affects or on-mount tasks.

```suggestion
	useEffect( () => {
```